manifests/init.pp

```diff
@@ -134,6 +137,16 @@
     notify  => Service['rhsmcertd'],
   }
 
+  if $package_ensure == 'absent' {
+    file { $repo_filename:
+      ensure => 'file',
+    }
+  } else {
+    file { $repo_filename:
+      ensure => 'absent',
+    }
+  }
```

Comment:
Although not an expert on RedHat, how does this work? If `packages_ensure` is present puppet will remove `/etc/yum.repos.d/redhat.repo`. What's the reason behind it?